sherpa/astro/datastack/ds.py

```diff
@@ -108,47 +121,70 @@ def clear_stack(self):
+        -------
+        header : list of strings
+            Column names
+        data : list of lists
+            Each list is a row, corresponding to one dataset
 
-        # The observation-date (MJD) format keyword was MJD_OBS but
-        # has changed (CIAO 4.12) for Chandra data to MJD-OBS, so
-        # need to look for both. The idea is to use the new key
-        # and then fall back to the old one.
-        #
+        '''
+        tab = []
         for dataset in self.filter_datasets():
-            obsid = "N/A"
-            time = "N/A"
-            timekey = 'MJD-OBS'
+            row = [dataset['id'], dataset['data'].name]
             try:
                 hdr = dataset['data'].header
+                row.extend([hdr.get(k, '') for k in keys_to_display])
+            except AttributeError:
+                row.append([''] * len(keys_to_display))
+            tab.append(row)
 
-                try:
-                    obsid = hdr['OBS_ID']
-                except KeyError:
-                    pass
-
-                for key in ['MJD-OBS', 'MJD_OBS']:
-                    try:
-                        time = hdr[key]
-                        timekey = key
-                        break
-                    except KeyError:
-                        pass
+        if len(tab) == 0:
+            header = ['empty datastack']
+            data = numpy.array([])
+        else:
+            data = numpy.array(tab)
+            # Find columns where not all entries are identical
+            ind = ~numpy.all(data == data[0, :], axis=0)
+            # But always keep the first two (id, name)
+            ind[:2] = True
+
+            header = numpy.array(['id', 'name'] + keys_to_display)[ind]
+            data = data[:, ind]
+        return header, data
+
+    def _repr_html_(self):
+        """Return a HTML (string) representation of the stack
+        """
+        header, data = self._summary_for_display()
+        htab = html_table(header, data, rowcount=False,
+                          summary=f'datastack with {data.shape[0]} datasets')
+        return html_from_sections(self, htab)
 
-            except AttributeError:
-                pass
+    # QUS: should this use the logger instead of print?
+    def show_stack(self):
+        """Display basic information about the contents of the data stack.
 
-            print('{0}: {1} {2}: {3} {4}: {5}'.format(dataset['id'],
-                                                      dataset['data'].name,
-                                                      'OBS_ID', obsid,
-                                                      timekey, time))
+        A brief summary of each data set in the stack is printed to the
+        standard output channel. The information displayed depends on the
+        type of each data set.
+        """
+        header, data = self._summary_for_display()
+        print('|'.join(header))
+        for row in data:
+            print('|'.join(row))
```

Comment:
This has a different feel to the other `show_xxx` routines, which tend to use more spaces, but I think the existing routines have a varied enough approach that we don't have a consistent guide.

Reply:
[a general comment, not a request for changes]